web/src/app/monitors/monitors-list/monitors-list.component.ts

```diff
@@ -57,16 +60,26 @@ export class MonitorsListComponent implements OnInit, OnDestroy {
    * @param uid the uid of monitor which needs to be deleted
    */
   deleteMonitor(monitorUid: string): void {
-    this.monitors = this.monitors.filter((monitor: MonitorModel) => monitor.uid !== monitorUid);
-    this.saveMonitorSubscription = this.monitorService.saveMonitors(this.projectUid, this.monitors)
-      .subscribe(
-        () => this.router.navigate([`/projects/${this.projectUid}/monitors`]),
-        (error: any): any => this.snackBar.open(error.message, undefined, { duration: 5000 })
-      );
-
-    // When delete a monitor , deleting all its pings
-    this.monitorService
-      .deletePingsByMonitor(this.projectUid, monitorUid);
+    let dialogConfig: MatDialogConfig = new MatDialogConfig();
+    dialogConfig = {
+      width: '500px',
+      data: {
+        title: 'Delete Monitor',
+        content: 'Are you sure you want to delete?',
+      },
+    };
+    this.dialogRef = this.dialog.open(DialogConfirmationComponent, dialogConfig);
+    this.dialogRef.afterClosed().subscribe((result: boolean) => {
+      if (result === true) {
+        this.saveMonitorSubscription = this.monitorService.saveMonitors(this.projectUid, this.monitors)
+          .pipe(
+            take(1),
+            switchMap(() => this.monitorService.deletePingsByMonitor(this.projectUid, monitorUid)))
+          .subscribe(() => this.monitors = this.monitors.filter((monitor: MonitorModel) => monitor.uid !== monitorUid),
```

Comment:
the first function in the subscribe should be on a new line to match the error

Reply:
fixed